### ISSUE TITLE
Pact Consumer tests are failing starting with node 18.10 and 19

### ISSUE THREAD

**Issue by RaresAil** [pact-foundation/pact-js#1066]
### Software versions

- **OS**: MacOS Ventrua
- **Pact Node version**:  Pact Foundation 10.3.1
- **Node Version**: 18.10.x, 19.x

### Issue Checklist

Please confirm the following:

- [x] I have upgraded to the latest
- [x] I have the read the FAQs in the Readme
- [x] I have triple checked, that there are **no unhandled promises** in my code
- [x] I have set my log level to debug and attached a log file showing the complete request/response cycle

### Expected behaviour

All the tests to pass

### Actual behaviour

The tests fail after upgrading from node `18.8.0` to `18.10.0` even on `19.x`

### Steps to reproduce

Update to node 18.10 or 19 and run the Consumer tests

### Relevant log files

```
● Console

    console.error


      at Pact.Object.<anonymous>.Pact.verify (../node_modules/@pact-foundation/src/httpPact/index.ts:211:15)
      at Object.<anonymous> (../node_modules/jest-pact/dist/pactWith.js:12:45)

    console.error
      [31mPact verification failed![39m

      at Pact.Object.<anonymous>.Pact.verify (../node_modules/@pact-foundation/src/httpPact/index.ts:212:15)
      at Object.<anonymous> (../node_modules/jest-pact/dist/pactWith.js:12:45)

    console.error
      [31mTest failed for the following reasons:
      
        Mock server failed with the following mismatches:
      
      	0) The following request was expected but not received: 
      	    Method: POST
      	    Path: ...
      	    Headers:
      	      Accept: application/json
      	      Authorization: Bearer mock
      	      Content-Type: application/json
      	    Body: ...

      at Pact.Object.<anonymous>.Pact.verify (../node_modules/@pact-foundation/src/httpPact/index.ts:213:15)
      at Object.<anonymous> (../node_modules/jest-pact/dist/pactWith.js:12:45)

  ● Pact between .... returns the correct response

    socket hang up

      at Function.Object.<anonymous>.AxiosError.from (../node_modules/axios/lib/core/AxiosError.js:89:14)
      at RedirectableRequest.handleRequestError (../node_modules/axios/lib/adapters/http.js:533:25)
      at ClientRequest.eventHandlers.<computed> (../node_modules/follow-redirects/index.js:14:24)

  ● Pact between ... returns the correct response

    Pact verification failed - expected interactions did not match actual.

      at new VerificationError (../node_modules/@pact-foundation/pact/src/errors/verificationError.js:21:42)
      at Pact.Object.<anonymous>.Pact.verify (../node_modules/@pact-foundation/src/httpPact/index.ts:217:13)
      at Object.<anonymous> (../node_modules/jest-pact/dist/pactWith.js:12:45)

Failed to initialise global tracing subscriber - a global default trace dispatcher has already been set
```

**Comment by mefellows** [comment]
> @mefellows This is a thing that could be changed in the proxy to avoid it in the future. If you want to do this, you can just add a bit of middleware that has res.append('Connection', 'close');

Thanks Tim - awesome digging. That should be easy enough to do.

**Comment by TimothyJones** [comment]
I think this issue is. the same as the one reported at the bottom of #1060 - since the error is `socket hang up`

**Comment by baileyandy** [comment]
For some reason, none of the above workarounds are helping my failing tests. I have reverted to node 16 for now.

However, I did discoverer that this issue has [been known](https://github.com/pact-foundation/pact-jvm/tree/master/consumer#dealing-with-persistent-http11-connections-keep-alive) in the JVM world for a while (if that helps!)

**Comment by maconic** [comment]
I know I am lat to the game, but having a similar issue with https://vitest.dev and pure fetch API implementation. In node 16 the Pact test works, but with 18 I can see in the console 
`pact_mock_server::server_manager: Shutting down mock server with ID`
even before the http client code executes. 

Test looks like this:

`describe("Data contract", () => {
  it("Should get data", async () => {
    provider
      .given("placeholder")
      .uponReceiving("placeholder")
      .withRequest({
        method: "GET",
        path: "/api/resource",
        headers: { Accept: "*/*" },
      })
      .willRespondWith({
        status: 200,
        headers: { "Content-Type": "application/json" },
        body: {},
      })

    return provider.executeTest(async () => {
      const dtos = await getMyDtos()
      expect(dtos).not.toBeNull()
    })
  }, 20000)
})`

**Comment by TimothyJones** [comment]
As above, I recommend reading the breaking changes between those node versions and adjusting accordingly.

**Comment by baileyandy** [comment]
@TimothyJones many thanks for the tips

I have re-read the Node changes in more detail and my issue was indeed caused by a change to the resolution of `localhost`. I ended up also finding [this stackoverflow](https://stackoverflow.com/questions/72648487/node-js-16-17-changed-resolution-of-localhost) question on my quest. I am on OSX which I guess is resolving to IPv6 first. Switching to `127.0.0.1` has fixed my issue.

interestingly all my pacts were running in **jsdom** jest environment and passed with the localhost -> IP change. Switching to **node** jest environment caused everything to fail again unless I also added `httpAgent: new http.Agent({ keepAlive: false })` to the axios config.
